tests/instruments/test_instrument.py

```diff
@@ -23,13 +23,52 @@
 #
 
 
+from unittest import mock
+
 import pytest
 
 from pymeasure.instruments import Instrument
+from pymeasure.instruments.instrument import DynamicProperty
+from pymeasure.adapters import FakeAdapter
+from pymeasure.test import expected_protocol
 from pymeasure.instruments.fakes import FakeInstrument
 from pymeasure.instruments.validators import strict_discrete_set, strict_range, truncated_range
 
 
+class GenericInstrument(FakeInstrument):
+    #  Use truncated_range as this easily lets us test for the range boundaries
+    fake_ctrl = Instrument.control(
+        "", "%d", "docs",
+        validator=truncated_range,
+        values=(1, 10),
+        dynamic=True,
+    )
+    fake_setting = Instrument.setting(
+        "%d", "docs",
+        validator=truncated_range,
+        values=(1, 10),
+        dynamic=True,
+    )
+    fake_measurement = Instrument.measurement(
+        "", "docs",
+        values={'X': 1, 'Y': 2, 'Z': 3},
+        map_values=True,
+        dynamic=True,
+    )
+
+
+@pytest.fixture(scope="module")
```

Comment:
I think we can afford not having this in module scope, so it gets recreated for every test -- this will also help with test isolation, and you don't need those dels anymore..

Reply:
I solved it differently, but yes, we can remove it.

Reply:
Good :-)